src/space_partition_exclude.c

```diff
@@ -0,0 +1,157 @@
+#include <postgres.h>
+#include <nodes/parsenodes.h>
+#include <nodes/nodeFuncs.h>
+#include <nodes/makefuncs.h>
+#include <parser/parsetree.h>
+#include <parser/parse_coerce.h>
+#include <parser/parse_func.h>
+#include <parser/parse_collate.h>
+#include <parser/parse_oper.h>
+#include <optimizer/clauses.h>
+#include <catalog/pg_type.h>
+#include <optimizer/paths.h>
+
+#include "cache.h"
+#include "hypertable.h"
+#include "partitioning.h"
+#include "compat.h"
+
+#include "space_partition_exclude.h"
+#include "chunk.h"
+#include "hypercube.h"
+#include "hypertable.h"
+
+typedef struct SpaceExcludeCtx
+{
+	Query	   *parse;
+	Hypertable *hentry;
+} SpaceExcludeCtx;
+
+
+
+static inline Dimension *
+get_dimension_for_partition_column_var(Var *var_expr, SpaceExcludeCtx *context)
+{
+	Hypertable *ht = context->hentry;
+	RangeTblEntry *rte = rt_fetch(var_expr->varno, context->parse->rtable);
+	char	   *varname;
+
+	if (rte->relid != ht->main_table_relid)
+		return NULL;
+
+	varname = get_rte_attribute_name(rte, var_expr->varattno);
+
+	return hyperspace_get_dimension_by_name(ht->space, DIMENSION_TYPE_CLOSED, varname);
+}
+
+SpacePartitionExclude *
+space_partition_exclude_get(RestrictInfo *base_restrict_info, Query *query, Hypertable *hentry)
```

Comment:
Also, maybe it makes sense to follow the postgres terminology and return a `ChunkRestrictInfo` or `HypertableRestrictInfo` instead of `SpacePartitionExclude`?